test/csv/test_encodings.rb

```diff
@@ -256,12 +256,11 @@ def test_row_separator_detection_with_invalid_encoding
   end
 
   def test_invalid_encoding_row_error
-    csv = CSV.new("invalid,\xF8\r\nvalid,x\r\n".force_encoding("UTF-8"),
-                  encoding: "UTF-8")
     error = assert_raise(CSV::MalformedCSVError) do
-      csv.shift
+      CSV.parse("valid,x\r\ninvalid,\xF8\r\n".force_encoding("UTF-8"),
+                encoding: "UTF-8")
     end
```

Comment:
We should not change to use `CSV.parse`. It accepts a case that `CSV.new` raises an error.
We should use `csv.shift` twice in the `assert_raise` block.